File: data structures/stack/stack.js
class Stack {
  constructor() {
    this.storage = [];
  }

  length = () => {
    return this.storage.length;
  };

  push = (value) => {
    this.storage.push(value);
  };

  pop = () => {
    if (this.length() > 0) {
      return this.storage.pop();
    } else {
      console.error('Cannot pop from empty stack!');
      return null;
    }
  };
}

const s = new Stack();

s.push(1);
s.push(2);
s.push(3);

console.log('length of stack:', s.length());
console.log('items in stack: ', s.storage);

console.log('popped value: ', s.pop());
console.log('popped value: ', s.pop());
console.log('popped value: ', s.pop());
console.log('popped value: ', s.pop());

console.log('length of stack:', s.length());

File: data structures/stack/stack.py
class Stack:
    def __init__(self):
        self.storage = []

    def __len__(self):
        return len(self.storage)

    def push(self, value):
        self.storage.append(value)

    def pop(self):
        if len(self.storage) > 0:
            return self.storage.pop()
        else:
            print("Cannot pop from empty stack!")
            return None


s = Stack()

s.push(1)
s.push(2)
s.push(3)

print("length of stack:", len(s))
print("items in stack:", s.storage)

print("popped value:", s.pop())
print("popped value:", s.pop())
print("popped value:", s.pop())
print("popped value:", s.pop())

print("items in stack:", s.storage)
